### ISSUE TITLE
Implement AddScalar hook

### ISSUE THREAD

**Issue by petrbel** [Cognexa/cxflow#186]
Some variables, such as learning rate, are scalar, i.e. their shape doesn't depend on batch size. These variables can't be stated in the model outputs.

AddScalar hook should overcome this problem and add specific scalars to
```epoch_data['train'][variable_name]```

Such hook will be used in the beginning of the hook list.
